src/zjs_callbacks.c

```diff
@@ -315,6 +334,20 @@ void zjs_remove_callback(int32_t id)
     }
 }
 
+void zjs_signal_callback_args(int32_t id, void* args, uint32_t size)
+{
+    if (args && size) {
+        int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)args, (uint8_t)size / 4);
```

Comment:
Just be careful because one other weirdness is that even if you pass size 0, if there were *no* arguments, and it all you wrote was the type/value stuff, it will still pop it off. (It's still able to read even though you didn't supply a buffer basically.)